core-chart/templates/argocd/setup.yaml

```diff
@@ -139,13 +77,104 @@ spec:
         app.kubernetes.io/instance: {{ .Release.Name | quote }}
         helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
     spec:
-      restartPolicy: Never # OnFailure
+      restartPolicy: Never
       securityContext:
         runAsNonRoot: true
       serviceAccountName: "{{ .Release.Name }}-argocd-kubestellar-setup"
+      
+      # INIT CONTAINERS - Handle all the waiting logic to ensure proper sequencing
+      initContainers:
+      
+      # INIT CONTAINER 1: Wait for ArgoCD Pod Creation
+      - name: wait-argocd-pod-create
+        image: quay.io/kubestellar/kubectl:{{ .Values.KUBECTL_VERSION }}
+        command: 
+        - kubectl
+        - wait
+        - --for=create
+        - pod
+        - -l
+        - app.kubernetes.io/name=argocd-server
+        - -n
+        - {{ .Release.Namespace }}
+        - --timeout=300s
+        securityContext:
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+              - ALL
+      
+      # INIT CONTAINER 2: Wait for ArgoCD Pod Ready
+      - name: wait-argocd-pod-ready
+        image: quay.io/kubestellar/kubectl:{{ .Values.KUBECTL_VERSION }}
+        command:
+        - kubectl
+        - wait
+        - --for=condition=Ready
+        - pod
+        - -l
+        - app.kubernetes.io/name=argocd-server
+        - -n
+        - {{ .Release.Namespace }}
+        - --timeout=300s
+        securityContext:
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+              - ALL
+      
+      # INIT CONTAINER 3: Wait for ArgoCD Secret Creation
+      - name: wait-argocd-secret-create
+        image: quay.io/kubestellar/kubectl:{{ .Values.KUBECTL_VERSION }}
+        command:
+        - kubectl
+        - wait
+        - --for=create
+        - secret/argocd-initial-admin-secret
+        - -n
+        - {{ .Release.Namespace }}
+        - --timeout=300s
+        securityContext:
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+              - ALL
+      
+      # INIT CONTAINER 4-N: Wait for Each Control Plane (Dynamic Loop)

```

Comment:
I see a loop but it is not dynamic. The range of the loop is fixed in the "values" of the chart instance.